test/integration/test_timeout.py

```diff
@@ -0,0 +1,214 @@
+        self._mock_out_cloudant_request()
+
+        testcases = self._defineTestCases()
+
+        for i, tc in enumerate(testcases):
+            tc['set_timeout'](self._custom_timeout)
+
+            # Call the server
+            self.cloudant.get_server_information()
+
+            # Assert timeout is set in the server request
+            req_args=self._get_request_arguments(i)
+            tc['assert_func'](req_args)
+
+    def test_timeout_cloudantv1_sessionauth(self, timeout_mock_response=_timeout_mock_response):
+        session_auth = CouchDbSessionAuthenticator('name', 'psw')
+        self.cloudant = CloudantV1(
+            authenticator=session_auth,
+        )
+        self.cloudant.set_service_url("http://cloudant.example")
+
+        # Mock out authentication
+        orig_request = requests.request
+        requests.request = Mock(return_value=timeout_mock_response)
+
+        # Mock out request response
+        self._mock_out_cloudant_request()
+
+        testcases = self._defineTestCases()
+
+        for i, tc in enumerate(testcases):
+            tc['set_timeout'](self._custom_timeout)
+
+            # Call the server
+            self.cloudant.get_server_information()
+
+            # Assert timeout is set to the authenticator
+            auth_args = self._get_authenticate_arguments()
+            self._assert_default_timeout_setting(auth_args)
+
+            # Assert timeout is set in the server request
+            req_args=self._get_request_arguments(i)
+            tc['assert_func'](req_args)
+
+        # Set back requests.request
+        requests.request = orig_request
+
+    def test_timeout_cloudantv1_iamauth(self, timeout_mock_response=_timeout_mock_response):
+        authenticator = IAMAuthenticator('apikey')
+        self.cloudant = CloudantV1(
+            authenticator=authenticator
+        )
+
+        # Mock out authentication
+        orig_request = requests.request
+        requests.request = Mock(return_value=timeout_mock_response)
+        self.cloudant.authenticator.token_manager._save_token_info=Mock()
+        self.cloudant.authenticator.token_manager.refresh_time = int(time.time()) + 200
+
+
+        # Mock out request response
+        self._mock_out_cloudant_request()
+
+        testcases = self._defineTestCases()
+
+        for i, tc in enumerate(testcases):
+            tc['set_timeout'](self._custom_timeout)
+
+            # Call the server
+            self.cloudant.get_server_information()
+
+            # Assert timeout is set to the authenticator
+            auth_args = self._get_authenticate_arguments()
+            self._assert_default_timeout_setting(auth_args)
+
+            # Assert timeout is set in the server request
+            req_args=self._get_request_arguments(i)
+            tc['assert_func'](req_args)
+
+            # Set expire time to not authenticate again
+            self.cloudant.authenticator.token_manager.expire_time = int(time.time()) + 200
```

Comment:
Should we mention that `200` is in seconds?

Reply:
I've introduced a new function: `get_current_time_plus_two_minute` in 7c0ccbf. WDYT?